include/eve/module/complex/complex.hpp

```diff
@@ -0,0 +1,145 @@
+    {
+      return get<0>(EVE_FWD(z));
+    }
+
+    EVE_FORCEINLINE friend decltype(auto) tagged_dispatch(eve::tag::imag_, like<complex> auto && z)
+    {
+      return get<1>(EVE_FWD(z));
+    }
+
+    //==============================================================================================
+    // Functions support
+    //==============================================================================================
+    template<typename Tag, like<complex> Z>
+    EVE_FORCEINLINE friend  auto  tagged_dispatch(Tag const& tag, Z const& z) noexcept
+                            ->    decltype(detail::complex_unary_dispatch(tag, z))
+    {
+      return detail::complex_unary_dispatch(tag, z);
+    }
+
+    template<typename Tag, decorator D, like<complex> Z>
+    EVE_FORCEINLINE friend  auto  tagged_dispatch(Tag const& tag, D const& d, Z const& z) noexcept
+                            ->    decltype(detail::complex_unary_dispatch(tag, d, z))
+    {
+      return detail::complex_unary_dispatch(tag, d, z);
+    }
+
+    template<typename Tag, typename Z1, typename Z2>
+    requires(like<Z1,complex> || like<Z2,complex>)
+    EVE_FORCEINLINE friend  auto  tagged_dispatch ( Tag const& tag
+                                                  , Z1 const& z1, Z2 const& z2
+                                                  ) noexcept
+                            ->    decltype(detail::complex_binary_dispatch(tag, z1, z2))
+    {
+      return detail::complex_binary_dispatch(tag, z1, z2);
+    }
+
+    //==============================================================================================
+    // Operator +
+    //==============================================================================================
+    EVE_FORCEINLINE friend auto operator+(like<complex> auto const& z) noexcept { return z; }
+
+    template<like<complex> Z1, like<complex> Z2>
+    EVE_FORCEINLINE friend auto& operator+= (Z1& self, Z2 const& o) noexcept
+    {
+      real(self) += real(o);
+      imag(self) += imag(o);
+      return self;
+    }
+
+    EVE_FORCEINLINE friend auto& operator+=(like<complex> auto& self, callable_i_ const&) noexcept
+    {
+      imag(self)++;
+      return self;
+    }
+
+    template<typename Z>
+    EVE_FORCEINLINE friend auto& operator+=(like<complex> auto& self, Z o) noexcept
+    requires(like<Z,Type> || std::convertible_to<Z,Type>)
+    {
+      real(self) += o;
+      return self;
+    }
+
+    template<like<complex> Z1, real_value Z2>
+    EVE_FORCEINLINE friend  auto operator+(Z1 const& lhs, Z2 const& rhs) noexcept
+    requires(requires(as_wide_as_t<Z1,Z2> t) { t += rhs; })
+    {
+      as_wide_as_t<Z1,Z2> that{lhs};
+      return that += rhs;
+    }
+
+    template<real_value Z1, like<complex> Z2>
+    EVE_FORCEINLINE friend  auto operator+(Z1 const& lhs, Z2 const& rhs) noexcept
+    requires(requires(as_wide_as_t<Z2,Z1> t) { t += lhs; })
+    {
+      return rhs + lhs;
```

Comment:
I prefer x and y after Stepanov

Reply:
instead of lhs/rhs ?

Reply:
yep. I did that in a few places. You can stick with lhs, rhs if you want. Noone is ever consistent about this.

Reply:
Will do